l1-contracts/src/core/libraries/validator-selection/ValidatorSelectionLib.sol

```diff
@@ -194,23 +205,35 @@ library ValidatorSelectionLib {
     ValidatorSelectionStorage storage store = getStorage();
     EpochData storage epoch = store.epochs[_epochNumber];
 
-    if (epoch.sampleSeed != 0) {
-      uint256 committeeSize = epoch.committee.length;
-      if (committeeSize == 0) {
-        return new address[](0);
-      }
-      return epoch.committee;
+    // If no committee has been stored, then we need to setup the epoch
+    uint256 committeeSize = epoch.committee.length;
+    if (committeeSize == 0) {
+      // This will set epoch.committee and the next sample seed in the store, meaning epoch.commitee on the line below will be set (storage reference)
+      setupEpoch(_stakingStore, _epochNumber);
     }
+    return epoch.committee;
+  }
 
-    // Allow anyone if there is no validator set
-    if (_stakingStore.attesters.length() == 0) {
-      return new address[](0);
-    }
+  function setSampleSeedForEpoch(Epoch _epoch) internal {
+    ValidatorSelectionStorage storage store = getStorage();
+    uint32 epoch = Epoch.unwrap(_epoch).toUint32();
 
-    // Emulate a sampling of the validators
-    uint256 sampleSeed = getSampleSeed(_epochNumber);
+    // Check if the latest checkpoint is for the next epoch
```

Comment:
If we set the initial using the constructor as well it might simplify us slightly here as well as we won't have the case where the last checkpoint does not exists so one less branch to deal with.